src/main/java/explorableviz/transparenttext/paragraph/Literal.java

```diff
@@ -1,22 +1,25 @@
 package explorableviz.transparenttext.paragraph;
 
 import explorableviz.transparenttext.variable.Variables;
-
 import static explorableviz.transparenttext.Program.replaceVariables;
 
 public class Literal extends TextFragment {
 
-    public Literal(String value) {
+    private final SelectedRegion selectedRegion;
+    public Literal(String value, SelectedRegion selectedRegion) {
         super(value);
+        this.selectedRegion = selectedRegion;
     }
-
     @Override
-    public Object clone() {
-        return new Literal(getValue());
+    public TextFragment replace(Variables computedVariables) {
```

Comment:
Where is this method used? (I thought we no longer had to pass `Variables` around.)